### PR TITLE
Migrate to NPM zip module

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3245.
Fixes: #3248.
Fixes: #3247.
Fixes: #3246.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8183/

**Comment by huwshimi** [comment]
QA:
- drop a charm zip on the canvas
- the local charm inspector should appear
- click to deploy the charm
- it should deploy

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8185/

File: jujugui/static/gui/src/app/init/test-endpoint-utils.js [hunk 1]
```diff
@@ -393,7 +391,7 @@ xdescribe('Endpoints map handlers', function() {
     controller.endpointsMap = {};
   });
 
-  afterEach(function(done) {
+  afterEach(function() {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/init/test-endpoint-utils.js]: Yes, seriously, turns out this is why all the tests were timing out :).

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/juju-gui/8187/

File: jujugui/static/gui/src/app/init/zip-utils.js [hunk 2]
```diff
@@ -30,16 +20,15 @@ if (window.juju_config && window.juju_config.staticURL) {
   @param {Function} errback A function called if any errors occur.
 */
 var getEntries = function(file, callback, errback) {
```
Comment (frankban) [inline jujugui/static/gui/src/app/init/zip-utils.js]: const?

File: jujugui/static/gui/src/app/init/zip-utils.js [hunk 3]
```diff
@@ -65,8 +54,7 @@ zipUtils.splitPath = splitPath;
   see juju-core/state/apiserver/charms.go:findArchiveRootDir.
 
   @method findCharmEntries
-  @param {Array} allEntries The list of all the zip entries
-    (see http://gildas-lormeau.github.io/zip.js/core-api.html#zip-entry).
+  @param {Array} allEntries The object of all the zip entries.
```
Comment (frankban) [inline jujugui/static/gui/src/app/init/zip-utils.js]: What does "The object of all the zip entries" mean?

File: jujugui/static/gui/src/app/init/zip-utils.js [hunk 4]
```diff
@@ -84,12 +72,8 @@ var findCharmEntries = function(allEntries) {
     'metadata.yaml': 'metadata',
     'revision': 'revision'
   };
-  allEntries.forEach(function(entry) {
-    if (entry.directory) {
```
Comment (frankban) [inline jujugui/static/gui/src/app/init/zip-utils.js]: So we no longer have directories returned?

**Comment by hatched** [comment]
I'm going to merge this to allow for qa but plz address @frankban 's comments in a follow-up if necessary.

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**